### PR TITLE
Rename stack commands to package commands

### LINKED ISSUE #54: update command names to use "package" instead of "stack" per updated terminology
image

### PR DESCRIPTION
**Pull request by hasheddan**
Fixes #54 

Renames all `kubectl crossplane stack` commands to `kubectl crossplane package` commands.

### CODE REVIEW

**Comment by hasheddan** [comment]
Please wait for me to merge after reviews are complete so I can coordinate other changes :+1:

File: README.md [hunk 1]
```diff
@@ -51,7 +51,7 @@ Or, if you customized the installation prefix:
 
 ```
 PREFIX=/thing
-rm "${PREFIX}"/bin/kubectl-crossplane-stack-*
+rm "${PREFIX}"/bin/kubectl-crossplane-package-*
```
Comment (prasek) [inline README.md]: might want to keep the old command as well for uninstalling older versions

**Review by prasek — APPROVED** [review]
Awesome, thanks @hasheddan!

File: README.md (hunk 1 above)
Comment (hasheddan) [reply README.md]: Good idea! Will update :)

File: README.md [hunk 2]
```diff
@@ -64,16 +64,37 @@ make uninstall
...
 
 ```
-kubectl crossplane stack init 'myname/mysubname'
-kubectl crossplane stack build
-kubectl crossplane stack publish
-kubectl crossplane stack install 'myname/mysubname'
-kubectl crossplane stack generate-install 'myname/mysubname' | kubectl apply --namespace mynamespace -f -
-kubectl crossplane stack list
-kubectl crossplane stack uninstall 'myname-mysubname'
+kubectl crossplane package init 'myname/mysubname'
```
Comment (jbw976) [inline README.md:70]: i have a feeling this list is out of date with the current commands, but i wouldn't hold this PR for that. Let's keep this PR focused on strict stack -> package naming change 👍

File: bin/kubectl-crossplane-package-init [hunk 3]
```diff
@@ -188,25 +188,25 @@ RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o manager
...
+  cat > package.env <<EOF
 # Image URL to use all building/pushing image targets
-STACK_IMG ?= ${STACK_NAME}:latest
+PACKAGE_IMG ?= ${PACKAGE_NAME}:latest
 
 CRD_DIR ?= config/crd/bases
-STACK_PACKAGE_REGISTRY_SOURCE ?= config/stack/manifests
-LOCAL_OVERRIDES_DIR ?= config/stack/overrides
-CONFIG_SAMPLES_DIR ?= config/stack/samples
+PACKAGE_REGISTRY_SOURCE ?= config/package/manifests
```
Comment (jbw976) [inline bin/kubectl-crossplane-package-init:202]: did you walk through the quick start all the way to do an init and a build to make sure that flow works?

how about for existing repos that were scaffolded out by this `init` command? is their repo structure going to be updated too?

**Review by jbw976 — APPROVED** [review]
looks pretty good! package doesn't even look like a word to me anymore after reviewing this 🤓 

https://en.wikipedia.org/wiki/Semantic_satiation

File: bin/kubectl-crossplane-package-init (hunk 3 above)
Comment (hasheddan) [reply bin/kubectl-crossplane-package-init:202]: Yep I tested this out with a new repo. However, I don't think any of the crossplane repos currently use this full setup. Providers do this by hand and stacks / apps only use build and publish.

**Comment by jbw976** [comment]
Is all feedback incorporated and we're ready to merge this one @hasheddan?

**Comment by hasheddan** [comment]
@jbw976 Yes, although the `cicd` and each of the stacks / apps need to follow in quick succession because I believe the `cicd` setup is periodic, so if it runs before it is updated or each of the stacks are updated then it will fail. It would not be a big problem, but I would rather minimize that window.

**Pull request merged**